x-pack/plugins/siem/server/graphql/kpi_hosts/schema.gql.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import gql from 'graphql-tag';
+
+export const kpiHostsSchema = gql`
+  type KpiHostsData {
+    hosts: Float
+    installedPackages: Float
+    processCount: Float
+    authenticationSuccess: Float
+    authenticationFailure: Float
+    fimEvents: Float
+    auditdEvents: Float
+    winlogbeatEvents: Float
+    filebeatEvents: Float
+    sockets: Float
+    uniqueSourceIps: Float
+    uniqueDestinationIps: Float
+  }
+
+  extend type Source {
+    KpiHosts(id: String, timerange: TimerangeInput!, filterQuery: String): KpiHostsData
```

Comment:
Do you want a `!` at the end of `KpiHostsData`? Without the `!` the source resolver generated will accept a null:

``` typescript
export namespace SourceResolvers {
  export interface Resolvers<Context = SiemContext, TypeParent = Source> {
    KpiHosts?: KpiHostsResolver<KpiHostsData | null, TypeParent, Context>;
}
```

Looking at the other resolvers under `SourceResolvers` in `/server/graphql/types.ts`, it seems most most only accept their explicit type (so with a `!` above).

Reply:
Good catch, thanks! It makes no sense if `KpiHosts` is null and accepted.